pygfx/utils/transform.py

```diff
@@ -735,16 +740,22 @@ def __init__(
         self.own = base
         self.own._set_wrapper(self)
 
-    def flag_update(self):
+    def flag_update(self, last_modified=None):
         """Signal that this transform has updated."""
-        self.last_modified = perf_counter_ns()
-        # this if-statement makes a massive difference for performance, don't remove it
+        # note: this function has been heavily micro-optimized
+        # please don't modify it carelessly
+        if last_modified is None:
+            last_modified = perf_counter_ns()
+        self.last_modified = last_modified
         if children := self.children:
             for child in children:
-                child.flag_update()
+                if child.children:
+                    child.flag_update(last_modified)
+                else:
+                    child.last_modified = last_modified
```

Comment:
With this, if the child has children, the child's `last_modified` is not set. Is that intentional?

Reply:
From what I see, the idea is to traverse faster by hopping two generation at a time instead of one, is that correct?

Reply:
* If the child has children, we recurse into the child's flag_update. That starts by setting self.last_modified, and then iterates over its children again. So we're not skipping anything.
* The idea is to avoid recursing into the leaf nodes, and just setting last_modified on them directly. Leaf nodes are usually the majority of the nodes in the scene graph so it makes an enormous difference.

Reply:
Right, with hopping I meant the recursion step. We recurse less, but handle two generation at each recursion.

I see how this should set `last_modified` on all descendants when this is the second recursion. But if this is the initial call, the `child` does not get its  `last_modified` set.

Reply:
> We recurse less, but handle two generation at each recursion.

But we don't! We only do this at the leaf level.

Reply:
Aaah! I see what I missed now. I somehow understood that `if child.children` it was going to call `flag_update` on *those* children, which is not what's happening. 

Sorry for the confusion!